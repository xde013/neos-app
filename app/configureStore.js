/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});
    /* eslint-enable */
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Hot Reloadable Reducers
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers));
    });
  }

  return store;
}
