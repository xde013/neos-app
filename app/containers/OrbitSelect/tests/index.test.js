/* eslint-disable react/jsx-props-no-spreading */
/**
 *
 * Tests for OrbitSelect
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { OrbitSelect, mapDispatchToProps } from '../index';

describe('<OrbitSelect />', () => {
  let store;
  let props;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  beforeEach(() => {
    props = {
      orbits: [],
      setFilter: jest.fn(),
      filter: '',
    };
  });
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <OrbitSelect {...props} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <OrbitSelect {...props} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('Props', () => {
    describe('setFilter', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.setFilter).toBeDefined();
      });
    });
  });
});
