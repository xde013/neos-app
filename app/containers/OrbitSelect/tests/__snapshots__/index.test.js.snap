// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrbitSelect /> Should render and match the snapshot 1`] = `
<div
  class="MuiFormControl-root makeStyles-formControl-74"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-simple-select-placeholder-label-label demo-simple-select-placeholder-label"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      id="demo-simple-select-placeholder-label"
      role="button"
      tabindex="0"
    >
      <em>
        All
      </em>
    </div>
    <input
      type="hidden"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-137 MuiOutlinedInput-notchedOutline"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend-138"
        style="width: 0.01px;"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    class="MuiFormHelperText-root MuiFormHelperText-contained"
  >
    Filter by Orbit
  </p>
</div>
`;
